File: 6-Patron/python/BendicionBotasJungla.py
import Botas


class BendicionBotasJungla(Botas):
    def __init__(self, habilidad):
        super().__init__()
        self.__habilidad = habilidad

    @property
    def habilidad(self):
        return self.__habilidad

    @habilidad.setter
    def habilidad(self, habilidad):
        self.__habilidad = habilidad

    def activarHabilidadRetribucion(self):
        self.__habilidad = 'Retribucion del angel caido'

    def verBotas(self):
        print(f"Habilidad: {self.__habilidad} ")

File: 6-Patron/Java/Bendicion_BotasJungla.java
package patron6;

public class Bendicion_BotasJungla extends Botas {
    private String Habilidad;

    public Bendicion_BotasJungla(String Habilidad) {
        this.Habilidad = Habilidad;
    }

    public String getHabilidad() {
        return Habilidad;
    }

    public void setHabilidad(String Habilidad) {
        this.Habilidad = Habilidad;
    }
    
    public void activarHabilidadRetribucion(){
        this.Habilidad = "Retribucion del angel caido";
    }
    
    @Override
    public void verBotas() {
        System.out.println("Habilidad: "+this.Habilidad);
    }
    
}
